### ISSUE TITLE
exa -l fails on latest Ubuntu upgrade 21.10 Gnome 3.38 X11

### ISSUE THREAD

**Issue by beshad** [ogham/exa#914]
been using exa for quite a while and only since last week after upgrading to Ubuntu 21.10 now i can't run "exa -l" or "exa --long:

`$ exa -l
thread 'main' panicked at 'index out of bounds: the len is 0 but the index is 0', /checkout/src/liballoc/vec.rs:1555:10
note: Run with `RUST_BACKTRACE=1` for a backtrace.
Aborted (core dumped)
`

other flags work fine, only the --long flag fails. any idea?

**Comment by ariasuni** [comment]
That is weird. Could you run `RUST_BACKTRACE=1 exa -l`? Does it happen in every directory?

**Comment by beshad** [comment]
yes it happens in every directory 

```
$ RUST_BACKTRACE=1 exa -l
thread 'main' panicked at 'index out of bounds: the len is 0 but the index is 0', /checkout/src/liballoc/vec.rs:1555:10
stack backtrace:
   0: <unknown>
   1: <unknown>
   2: <unknown>
   3: <unknown>
   4: <unknown>
   5: <unknown>
   6: <unknown>
   7: <unknown>
   8: <unknown>
   9: <unknown>
  10: <unknown>
  11: __libc_start_main
  12: <unknown>
Aborted
```
exa -l works fine on my other distros and worked here too until i upgraded to 21.10

**Comment by gek169** [comment]
Maybe it's a breaking change in Rust?

**Comment by beshad** [comment]
solved the problem by removing version 0.8.0 and reinstalling v0.10.1 from the repo here. 

`apt install exa` installs v0.9 but i downloaded version 0.10, unzipped it and copied exa* to /usr/local/bin/ and now i have a working v0.10 

i had initially followed the instruction from this website over a year ago (https://ourcodeworld.com/articles/read/832/how-to-install-and-use-exa-a-modern-replacement-for-the-ls-command-in-ubuntu-16-04) so maybe that wasn't the correct method?! 

all sorted now

**Comment by ariasuni** [comment]
This tutorial asks to install Rust but it’s not necessary, so that’s kind of weird to be honest. Also it doesn’t give the instructions to benefit from shell completions and man pages, but that’s another thing… But yeah it’s relatively old and makes you download exa 0.8, so that was the problem.
